File: agencia/iceibank/src/main/java/com/dadm/iceibank/util/RelogioLamport.java
package com.dadm.iceibank.util;

public class RelogioLamport {
    private int contador = 0;

    public synchronized int eventoLocal() {
        contador += 1;
        return contador;
    }

    public synchronized int aoEnviar() {
        contador += 1;
        return contador;
    }

    public synchronized int aoReceber(int timeStampRecebido) {
        contador = Math.max(contador, timeStampRecebido) + 1;
        return contador;
    }
}

File: agencia/relogio_lamport.py
class RelogioLamport:
    def __init__(self):
        self.contador = 0

    def evento_local(self):
        self.contador += 1
        return self.contador

    def ao_enviar(self):
        self.contador += 1
        return self.contador

    def ao_receber(self, timestamp_recebido):
        self.contador = max(self.contador, timestamp_recebido) + 1
        return self.contador